.github/workflows/release-pr.yml

```diff
@@ -0,0 +1,64 @@
+name: Changelog
+on:
+  push:
+    branches:
+      - release-*
+
+jobs:
+  release-notes-pr:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Extract branch
+        id: extract_branch
+        run: echo "branch=${GITHUB_REF#refs/heads/}" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_REF: ${{ github.ref_name }}
+
+      - name: Extract tag from branch name
+        id: extract_tag
+        run: echo "tag=${BRANCH#release-}" >> "$GITHUB_OUTPUT"
+        env:
+          BRANCH: ${{ steps.extract_branch.outputs.branch }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Get latest release
+        id: get_latest_release
+        run: |
+          latest_release=$(gh release view --json tagName --jq .tagName)
+          echo $latest_release
+          echo "tag=$latest_release" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate release notes
+        id: generate_release_notes
+        # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release
+        run: |
+          response=$(curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/yeagerai/genlayer-simulator/releases/generate-notes \
+            -d '{"tag_name":"${{ steps.extract_tag.outputs.tag }}","target_commitish":"${{ steps.extract_branch.outputs.branch }}","previous_tag_name":"${{ steps.get_latest_release.outputs.tag}}"}' )
+          echo $response
+
```

Comment:
Can you use https://github.com/marketplace/actions/release-notes-generator instead of this?

Reply:
First time I see that one, but I've researched a lot about tools to automate this and I didn't find anyone that would satisfy our needs. All of them had little to no flexibility to customize them.

For example, the one you are sharing here only works with "milestones"

Do you know how to use it for this use case?
We need to create release notes when there's a release branch created, and add those notes to a PR body